node/resolvers/profile/profileResolver.ts

```diff
@@ -37,71 +14,74 @@ export const customFieldsFromGraphQLInput = (customFieldsInput) => compose(
+  return {
     headers: {
-      'Proxy-Authorization': ctx.authToken,
-      vtexidclientautcookie: ctx.authToken,
+      'Proxy-Authorization': context.authToken,
+      vtexidclientautcookie: context.authToken,
     },
   }
+}
 
-  const profileURL = paths.profile(ctx.account).filterUser(user, customFields)
-  const profileData = await http.get(profileURL, config).then<any>(pipe(prop('data'), head))
+// This fix is necessary because some addresses were saved with the profile.id as userId instead of the actual profile.userId
+// TO DO: Remove this on the Future
+const fixAddresses = async (context, currentProfile: CurrentProfile, profileId: String) => {
+  const config = getProfileHeadersConfig(context)
+  const addressURL = paths.profile(context.account).filterAddress(profileId)
 
-  if (profileData && profileData.id) {
-    profileData.customFields = pickCustomFieldsFromData(customFields, profileData)
+  const addressesToFix = await http.get(addressURL, config).then(prop('data'))
 
-    profileData.profilePicture = profileData.profilePicture
-      && `//api.vtex.com/${ctx.account}/dataentities/CL/documents/${profileData.id}/profilePicture/attachments/${profileData.profilePicture}`
+  const fixed = addressesToFix.map(async (address) => {
+    http.patch(paths.profile(context.account).address(address.id),{ userId: currentProfile.userId }, config)
+    address.userId = currentProfile.userId
 
-    const addressURL = paths.profile(ctx.account).filterAddress(profileData.id)
-    const address = profileData && await http.get(addressURL, config).then(prop('data'))
+    return address
+  })
 
-    const paymentsURL = paths.profile(ctx.account).payments(profileData.userId)
-    const { data: paymentsRawData } = await http.get(paymentsURL, config)
-    let payments = null
+  return await Promise.all(fixed)
+}
 
-    if(paymentsRawData) {
-      const availableAccounts = JSON.parse(paymentsRawData.paymentData).availableAccounts
-      payments = availableAccounts.map((account) => {
-        const {bin, availableAddresses, accountId, ...cleanAccount} = account
-        const accountAddress = address.find(
-          (addr) => addr.addressName === availableAddresses[0]
-        )
-        return {...cleanAccount, id: accountId, address: accountAddress}
-      })
-    }
+export const getProfileData = async (context, currentProfile: CurrentProfile, args) => {
+  const config = getProfileHeadersConfig(context)
+  const { customFields } = args
 
-    return merge({ payments, address, cacheId: user }, profileData)
-  }
-  return {
-    cacheId: user,
-    email: user
+  const profileURL = paths.profile(context.account).filterUser(currentProfile.email, customFields)
+  const profileData = await http.get(profileURL, config).then<any>(pipe(prop('data'), head))
+
+  if(profileData && profileData.userId === currentProfile.userId) {
+    profileData.address = await getAddresses(context, currentProfile, profileData.id)
```

Comment:
```suggestion
    return {
      ...profileData,
      address: await getAddresses(context, currentProfile, profileData.id),
    }
```

Reply:
I rather keep it as it is right now, I don't see a real gain with this approach